src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -180,6 +199,61 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     }
   }
 
+  def syncInternalCreateCluster(userEmail: WorkbenchEmail,
+                                serviceAccountInfo: ServiceAccountInfo,
+                                googleProject: GoogleProject,
+                                clusterName: ClusterName,
+                                clusterRequest: ClusterRequest): Future[Cluster] = {
+    // Check if the google project has an active cluster with the same name. If not, we can create it
+    dbRef.inTransaction { dataAccess =>
+      dataAccess.clusterQuery.getActiveClusterByName(googleProject, clusterName)
+    } flatMap {
+      case Some(existingCluster) =>
+        throw ClusterAlreadyExistsException(googleProject, clusterName, existingCluster.status)
+      case None =>
+        val augmentedClusterRequest = addClusterDefaultLabels(
+          serviceAccountInfo, googleProject, clusterName, userEmail, clusterRequest)
+        val machineConfig = MachineConfigOps.create(clusterRequest.machineConfig, clusterDefaultsConfig)
+        val autopauseThreshold = calculateAutopauseThreshold(
+          clusterRequest.autopause, clusterRequest.autopauseThreshold)
+        val initialCluster = Cluster.create(
+          clusterRequest, userEmail, clusterName, googleProject,
+          serviceAccountInfo, machineConfig, dataprocConfig.clusterUrlBase, autopauseThreshold)
+
+        val attemptToSaveClusterInDb: Future[Cluster] = dbRef.inTransaction(_.clusterQuery.save(initialCluster))
+
+        // For the success case, register the following callbacks...
+        attemptToSaveClusterInDb foreach { cluster =>
+          // Notify the auth provider and kick off the cluster creation on the Google Dataproc side
+          asyncClusterCreate(userEmail, serviceAccountInfo, googleProject, clusterName, augmentedClusterRequest)
+            .onComplete {
+              case Success(_) =>
+                // Finally, notify the cluster monitor that the cluster has been created
+                clusterMonitorSupervisor ! ClusterCreated(cluster, clusterRequest.stopAfterCreation.getOrElse(false))
+              case Failure(_) =>
+                // If we didn't succeed, createGoogleCluster removes resources in Google but
+                // we also need to notify our auth provider that the cluster has been deleted.
+                // We won't wait for that deletion, though.
+                authProvider.notifyClusterDeleted(userEmail, userEmail, googleProject, clusterName)
```

Comment:
If the async portion fails we should report the error to the user, probably by setting the Cluster status to Error in the DB and adding a `ClusterError` record.